server/aesdsocket.c

```diff
@@ -61,43 +93,271 @@ void *get_in_addr(struct sockaddr *sa)
 // graceful exit
 void exit_handler(int signum)
 {
+    if (signum == SIGINT || signum == SIGTERM)
+    {
+        syslog(LOG_DEBUG, "Caught signal, exiting");
 
-    syslog(LOG_DEBUG, "Caught signal, exiting");
+        if(0 != shutdown(server_fd, SHUT_RDWR))
+        {
+            syslog(LOG_DEBUG, "shutdown failed");
+        }
+    }
 
-    close(fd);	
-	close(server_fd);
-    close(client_fd);
-    closelog();
+}
 
-    // Delete file
-    if(remove(FILE_NAME) < 0) 
+void timer_handler(int signum)
+{
+
+    int ret;
+    int write_size;
+
+    time_t rawtime;
+    struct tm *info;
+    char time_buffer[TIME_BUFFER_SIZE];
+    char print_buffer[PRINT_BUFFER_SIZE];
+
+    time( &rawtime );
+    info = localtime( &rawtime );
+    strftime(time_buffer, TIME_BUFFER_SIZE, "%Y/%m/%d-%H:%M:%S", info);
+    sprintf(print_buffer, "timestamp:%s\n", time_buffer);  
+
+
+    write_size = strlen(print_buffer);
+
+    if ( 0 != pthread_mutex_lock(&mutex_lock))
```

Comment:
@SundarKrishnakumar is this safe to do in a signal handler?  What would happen if a signal occurred while your thread was holding `mutex_lock`?